src/hooks/useCheckRevision.ts

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+import useAlert from './useAlert';
+import useFocusInput from './useFocusInput';
+
+const useCheckRevision = () => {
+  const { useSetAlert } = useAlert();
+  const { handleChildClick } = useFocusInput();
+  const [checked, setChecked] = React.useState<string[]>([]);
+
+  const handleToggle = (e: React.MouseEvent) => {
+    const revisionID = (e.currentTarget as HTMLElement).id;
+    const currentIndex = checked.indexOf(revisionID);
+    const newChecked = [...checked];
+
+    handleChildClick(e);
+
+    if (checked.length < 4 && currentIndex === -1) {
+      newChecked.push(revisionID);
+    } else if (currentIndex !== -1) {
+      newChecked.splice(currentIndex, 1);
+    } else {
+      useSetAlert('Maximum 4 Revisions', 'warning');
```

Comment:
is this alert timed? or it will remain on the page until it has been addressed?

Reply:
as it is now, it will stay until dismissed.